backend/gsr_booking/api_wrapper.py

```diff
@@ -489,38 +230,76 @@ def book_room(self, rid, start, end, user, test=False):
 
         res_json = response.json()
         # corrects keys in response
-        if "error" not in res_json:
-            if "errors" in res_json:
-                errors = res_json["errors"]
-                if isinstance(errors, list):
-                    errors = " ".join(errors)
-                res_json["error"] = BeautifulSoup(
-                    errors.replace("\n", " "), "html.parser"
-                ).text.strip()
-                del res_json["errors"]
+        if "error" not in res_json and "errors" in res_json:
+            errors = res_json["errors"]
+            if isinstance(errors, list):
+                errors = " ".join(errors)
+            res_json["error"] = BeautifulSoup(
+                errors.replace("\n", " "), "html.parser"
+            ).text.strip()
+            del res_json["errors"]
         if "error" in res_json:
             raise APIError("LibCal: " + res_json["error"])
         return res_json
 
     def get_reservations(self, user):
+        pass
 
-        reservations = Reservation.objects.filter(
-            creator=user, end__gte=timezone.localtime(), is_cancelled=False
-        )
-        group_gsrs = GSRBooking.objects.filter(
-            reservation__in=reservations, gsr__in=GSR.objects.filter(kind=GSR.KIND_LIBCAL)
-        )
+    def cancel_room(self, booking_id, user):
+        """Cancels room"""
+        # gsr_booking = get_object_or_404(GSRBooking, booking_id=booking_id)
+        # if gsr_booking:
+        #     if user != gsr_booking.user and user != gsr_booking.reservation.creator:
+        #         raise APIError("Error: Unauthorized: This reservation was booked by someone else.")
+        #     gsr_booking.is_cancelled = True
+        #     gsr_booking.save()
+        response = self.request("POST", f"{API_URL}/1.1/space/cancel/{booking_id}").json()
```

Comment:
nit: check status code first

Reply:
no need anymore if we check ok in the request first right